src/karmabot/commands/note.py

```diff
@@ -104,19 +106,20 @@ def _parse_note_cmd(text: str) -> Tuple[str, str]:
 
 
 def _get_notes_for_user(user: KarmaUser) -> list:
-    return (
-        db_session.create_session()
-        .query(KarmaNote)
-        .filter_by(user_id=user.user_id)
-        .all()
-    )
+    with database.session_manager() as session:
+        notes = session.query(KarmaNote).filter_by(user_id=user.user_id).all()
+    return notes
 
 
 def _note_exists(msg: str, user: KarmaUser) -> bool:
-    session = db_session.create_session()
-    q = session.query(KarmaNote).filter_by(note=msg, user_id=user.user_id)
-
-    return session.query(q.exists()).scalar()  # returns True or False
+    with database.session_manager() as session:
+        selected_note = session.query(KarmaNote).filter_by(

```

Comment:
Do we need that second query, can't we just get the boolean from `selected_note` directly?

Reply:
@pmayd should take a look. I just refactored the old one to use the new database :)

Reply:
ok not a big deal, @pmayd if you can take a look, also to this PR overall please? thanks

Reply:
I can try but my time is now very limited and this PR is really huge so I won't find the time to review it in detail, maybe at the weekend and for a couple of minutes at the evenings